sqlmesh/core/model/definition.py

```diff
@@ -966,14 +966,38 @@ def _data_hash_values(self) -> t.List[str]:
                 statement_exprs = [statement]
             else:
                 rendered = self._statement_renderer(statement).render()
-                if rendered is not None:
+                if self._is_metadata_statement(statement):
+                    continue
+                if rendered:
                     statement_exprs = rendered
                 else:
                     statement_exprs = [statement]
             data_hash_values.extend(gen(e) for e in statement_exprs)
 
         return data_hash_values
 
+    @property
+    def _additional_metadata(self) -> t.List[str]:
+        additional_metadata = super()._additional_metadata
+
+        for statement in (*self.pre_statements, *self.post_statements):
+            if self._is_metadata_statement(statement):
+                additional_metadata.append(gen(statement))
+
+        return additional_metadata
+
+    def _is_metadata_statement(self, statement: exp.Expression) -> bool:
+        if isinstance(statement, d.MacroDef):
+            return False
```

Comment:
should macro def always be metadata, there's no side effects

Reply:
yeah, I think we can do that